### PR TITLE
Make TLS mode default

### PR DESCRIPTION
**Pull request by lindnerby**
**Description:**

**Changes:**
* Remove option to decide between http and https, make TLS default
* Small refactorings:
    * Extract json serializable types into file
    * Inject deserializer as dependency to Handler instead of global var

**Related issue:**
* https://github.com/kyma-project/lifecycle-manager/issues/794

### CODE REVIEW

**Review by jeremyharisch — DISMISSED** [review]

File: runtime-watcher/internal/handler.go [hunk 1]
```diff
@@ -34,17 +33,17 @@ const (
...
+	Deserializer runtime.Decoder
 }
 
 type ServerParameters struct {
 	Port        int    // webhook server port
 	CACert      string // CA key used to sign the certificate
 	TLSCert     string // path to TLS certificate for https
 	TLSKey      string // path to TLS key matching for certificate
-	TLSServer   bool   // indicates if an HTTPS server should be created
 	TLSCallback bool   // indicates if KCP accepts HTTP or HTTPS requests
```
Comment (nesmabadr) [inline runtime-watcher/internal/handler.go:47]: Is there a reason you didn't remove the TLSCallback as well?

Comment (lindnerby) [reply runtime-watcher/internal/handler.go]: Of course, removed.

Comment (lindnerby) [reply runtime-watcher/internal/handler.go]: Seems it was introduced to make testing easier and bypass https for the test. Will try to introduce cert generation in the test

Comment (lindnerby) [reply runtime-watcher/internal/handler.go:47]: As discussed we will enable tls for the env test in a separate [issue](https://github.com/kyma-project/runtime-watcher/issues/129).

**Pull request merged**